colcon_ros_gradle/package_identification/ros_gradle.py

```diff
@@ -22,134 +22,25 @@ def __init__(self):  # noqa: D107
             PackageIdentificationExtensionPoint.EXTENSION_POINT_VERSION,
             '^1.0')
 
-    def identify(self, metadata):  # noqa: D102
-        if metadata.type is not None and metadata.type != 'gradle':
+    def identify(self, desc):  # noqa: D102
+        if desc.type is not None and desc.type != 'ros.gradle':
             return
 
-        build_gradle = metadata.path / 'build.gradle'
-        if not build_gradle.is_file():
+        # Call ROS package identification extention.
+        ros_extension = RosPackageIdentification()
+        ros_extension.identify(desc)
```

Comment:
This will modify `desc` and if the build type is a "common" one like `cmake` this extension actually overrides the `RosPackageIdentification` one which should not be the case.

Reply:
Yes, the `ros_extension` should only receive a copy. But at the end of the day the `desc` instance passed in as an argument to this extension needs to be populated / updated if this extension wants to "identify" the package. If it doesn't other extensions with lower priority will get a chance.

Reply:
Clear thanks !